ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java

```diff
@@ -280,6 +293,115 @@ public synchronized Container getContainerById(int accountId, int containerId) t
+      long startTimeMs = System.currentTimeMillis();
+      dataAccessor.getDatabaseConnection(false);
+      PreparedStatement getDatasetVersionStatement =
+          dataAccessor.getPreparedStatement(getDatasetVersionByNameSql, false);
+      DatasetVersionRecord result =
+          executeGetDatasetVersionStatement(getDatasetVersionStatement, accountId, containerId, datasetName, version);
+      dataAccessor.onSuccess(Read, System.currentTimeMillis() - startTimeMs);
+      return result;
+    } catch (SQLException e) {
+      dataAccessor.onException(e, Read);
+      throw e;
+    }
+  }
+
+  /**
+   * Execute insert dataset version statement to add a dataset version.
+   * @param statement the mysql statement to add dataset version.
+   * @param accountId the id for the parent account.
+   * @param containerId the id of the container.
+   * @param datasetName the name of the dataset.
+   * @param version the version of the dataset.
+   * @param expirationTimeMs the expiration time in milliseconds.
+   * @throws SQLException
+   */
+  private Dataset executeAddDatasetVersionStatement(PreparedStatement statement, int accountId, int containerId,
+      String accountName, String containerName, String datasetName, long version, long expirationTimeMs)
+      throws SQLException {
+    Dataset dataset = getDatasetMetadata(accountId, containerId, accountName, containerName, datasetName, expirationTimeMs);
+    statement.setInt(1, accountId);
+    statement.setInt(2, containerId);
+    statement.setString(3, datasetName);
+    statement.setLong(4, version);
+    if (expirationTimeMs != Infinite_Time) {
+      statement.setTimestamp(5, new Timestamp(expirationTimeMs));
+    } else {
+      if (dataset.getExpirationTimeMs() != Infinite_Time) {
+        statement.setTimestamp(5, new Timestamp(dataset.getExpirationTimeMs()));
+      } else {
+        statement.setTimestamp(5, null);
+      }
+    }
+    statement.executeUpdate();
+    return dataset;
+  }
+
+  /**
+   * Get the {@link Dataset} metadata which might be used for dataset version
+   * @param accountId the id for the parent account.
+   * @param containerId the id of the container.
+   * @param accountName the name of the account.
+   * @param containerName the name of the container.
+   * @param datasetName the name of the dataset.
+   * @param expirationTimeMs the expiration time in milliseconds.
+   * @return the {@link Dataset}
+   * @throws SQLException
+   */
+  private Dataset getDatasetMetadata(int accountId, int containerId, String accountName, String containerName,
+      String datasetName, long expirationTimeMs) throws SQLException {
+    Dataset dataset = getDataset(accountId, containerId, accountName, containerName, datasetName);
+    if (dataset.getExpirationTimeMs() != -1 && dataset.getExpirationTimeMs() < expirationTimeMs) {
+      throw new SQLException(
+          "This accountName " + accountName + " containerName: " + containerName + " datasetName: " + datasetName
```

Comment:
thanks, just for uniformity 
-> "accountName: " <- colon
->  and space before " already"

Reply:
done